chore(aws): add multi-region concept page (#3519)
camunda/camunda-docs: docs/self-managed/concepts/multi-region/dual-region.md

---
id: dual-region
title: "Dual-Region"
sidebar_label: "Dual-Region"
description: "Dual-Region allows you to run Camunda Platform connected in two regions."
---

import DualRegion from "./img/dual-region.svg";

## Introduction

Camunda 8 is compatible with a dual-region setup under certain [limitations](#limitations). This allows Camunda 8 to run in a mix of active-active and active-passive setups, resulting in an overall **active-passive** setup. The following will explore the concept, limitations, and considerations.

> You should get yourself familiar with the topic, the [limitations](#limitations) of Camunda 8 but also general [considerations](#considerations) on operating a dual-region setup.

Reviewer: ```suggestion
You should get familiar with the topic, the [limitations](#limitations) of the dual-region setup, and the general [considerations](#considerations) on operating a dual-region setup.
```

## Disclaimer

:::danger

> - Customers must develop and test operational procedures in non-production environments based on the framework steps outlined by Camunda.

Reviewer: ```suggestion
- Customers must develop and test operational procedures in non-production environments based on the framework steps outlined by Camunda, **before applying them in production setups**.
```

- Before advancing to production go-live, customers need to validate these procedures with Camunda.

:::

## Architecture

<DualRegion />

> The current architecture consists of the following supported components:

Reviewer: Let's add here that 2 regions mean 2 separate Kubernetes clusters able to communicate to each other (from the image, it's not immediately obvious)

- Zeebe (Workflow Engine)
- Elasticsearch (Database)
- WebApps
  - Operate
  - Tasklist

> Zeebe stretches across the regions due to its use of the [raft protocol](<https://en.wikipedia.org/wiki/Raft_(algorithm)>) allowing it to communicate and replicate data between all brokers across. Zeebe is exporting data to two Elasticsearch instances, one in each region. Operate and Tasklist will import the previously exported data and run per region.

Reviewer: ```suggestion
Zeebe stretches across the regions due to its use of the [Raft protocol](<https://en.wikipedia.org/wiki/Raft_(algorithm)>) allowing it to communicate and replicate data between all brokers across. Zeebe is exporting data to two Elasticsearch instances, one in each region. Operate and Tasklist will import the previously exported data and run per region.
```

In an **active-active** setup, multiple application instances run simultaneously in different regions, actively handling user requests. This allows for better load balancing and fault tolerance, as traffic can spread across regions. If one region fails, the workload can shift to another without causing disruptions.

In contrast, an **active-passive** setup designates one region as the main or active region where all user requests are processed. The other region remains on standby until needed, only becoming active if the primary region fails. This setup is easier to manage but may result in higher delays during failover events.

### User traffic

The overall system is **active-passive**, even though some components may be **active-active**. You will have to take care of the user traffic routing or DNS by yourself, and won't be considered further. You must select one region as the actively serving region and route the user traffic there. In case of a total region failure, you must route the traffic to the passive region yourself.

<!-- Should we provide some reading materials on how to tackle this? -->

### Components

> #### Zeebe ![active](https://img.shields.io/badge/active-green?style=for-the-badge)

Author: just a suggestion with shield.io (we would mirror it as local file to avoid injections if accepted).
Wanted to not having looking that triste.

Reviewer: not sure about the badge as a single word, I think it should be `active-active` or `active-passive`, just `active` or `passive` on their own are confusing.

Something like this:
https://img.shields.io/badge/active/active-green?style=for-the-badge

Or just using a text instead of a badge.

Author: I removed it for now, the bold active-active / active-passive in each first sentence of the components indicates it quite quickly.
Was just an experiment.

> Zeebe, as a workflow engine, is fully **active-active** and replicates data between its brokers. Due to its replication logic, it can handle an entire region failure without data loss. Requiring a proper partition and replication setup. Consider reading through the [cluster technical concept](./../../../components/zeebe/technical-concepts/clustering.md) to learn more about the [raft protocol](<https://en.wikipedia.org/wiki/Raft_(algorithm)>).

Reviewer: ```suggestion
Zeebe, as a workflow engine, is fully **active-active** and replicates data between its brokers. Due to its replication logic, it can handle an entire region failure without data loss. Requiring a proper partition and replication setup. Consider reading through the [cluster technical concept](./../../../components/zeebe/technical-concepts/clustering.md) to learn more about the [Raft protocol](<https://en.wikipedia.org/wiki/Raft_(algorithm)>).
```

Per region, one WebApp instance is required for the exporter to run.

This means that one instance will be actively serving traffic, while the other one is on standby.

##### Operate

In the event of a total primary region loss, the following data will be lost:

- Batch Operations <!-- Do we have a docs link for this? -->

##### Tasklist

In the event of a total primary region loss, the following data will be lost:

- Assignments of tasks

## Requirements

- Camunda 8
  - Minimum [helm chart version](https://github.com/camunda/camunda-platform-helm) **9.3+**
  - Minimum component images
    - Elasticsearch **8.9+**
    - Operate **8.5+**
    - Tasklist **8.5+**
    - Zeebe **8.5+**
    - Zeebe Gateway **8.5+**
- Two connected Kubernetes clusters
- Maximum latency of 100ms between the two Kubernetes clusters
- Open Ports between the two Kubernetes clusters
  - **9200** for Elasticsearch since Zeebe has to push data cross region
  - **26500** for the Zeebe gateway from client/worker communication
  - **26501** for the Zeebe brokers and Zeebe gateway communication
  - **26502** for the Zeebe brokers and Zeebe gateway communication

## Limitations

> - This is only about a dual-region setup and does not replicate any multi-region setup with more than two Kubernetes clusters

Reviewer: ```suggestion
- This guide only covers a dual-region Camunda 8 setup, installed in 2 Kubernetes clusters
```

Author: removed it altogether

>   - a multi-region setup of 3+ clusters will behave differently due to for example the raft protocol within Zeebe

Reviewer: ```suggestion
  - a multi-region setup of 3+ clusters will behave differently due to, for example, the Raft protocol used by Zeebe
```

(not even sure why we want to mention 3-region here tbh)

Author: so people don't get the wrong idea that just because it works for two, it will work for three.

Author: removed it altogether

You should get yourself familiar with the topic, the [limitations](#limitations) of Camunda 8 but also general [considerations](#considerations) on operating a dual-region setup.

## Disclaimer

:::danger

- Customers must develop and test operational procedures in non-production environments based on the framework steps outlined by Camunda.
- Before advancing to production go-live, customers need to validate these procedures with Camunda.

:::

## Architecture

<DualRegion />

The current architecture consists of the following supported components:

- Zeebe (Workflow Engine)
- Elasticsearch (Database)
- WebApps
  - Operate
  - Tasklist

Zeebe stretches across the regions due to its use of the [raft protocol](<https://en.wikipedia.org/wiki/Raft_(algorithm)>) allowing it to communicate and replicate data between all brokers across. Zeebe is exporting data to two Elasticsearch instances, one in each region. Operate and Tasklist will import the previously exported data and run per region.

> One of the regions will be considered **active** and the other one **passive**. User traffic must only reach the **active** region.

Author: we can define it but on just that picture doesn't matter since we disregard traffic

<!-- Should we provide some reading materials on how to tackle this? -->

### Components

#### Zeebe ![active](https://img.shields.io/badge/active-green?style=for-the-badge)

Zeebe, as a workflow engine, is fully **active-active** and replicates data between its brokers. Due to its replication logic, it can handle an entire region failure without data loss. Requiring a proper partition and replication setup. Consider reading through the [cluster technical concept](./../../../components/zeebe/technical-concepts/clustering.md) to learn more about the [raft protocol](<https://en.wikipedia.org/wiki/Raft_(algorithm)>).

#### Elasticsearch ![passive](https://img.shields.io/badge/passive-orange?style=for-the-badge)

We treat Elasticsearch as an **active-passive** component. While it may be possible to deploy it in active-active mode and stretch across regions, it's not officially documented by Elasticsearch. Such a setup generally brings its own challenges along that we can't control and may negatively impact the performance of Zeebe.

> We go with the approach of keeping an Elasticsearch per region and populating the data directly from the Zeebe exporter in each region.

Reviewer: ```suggestion
We go with the approach of having one Elasticsearch per region and configuring Zeebe to export the Elasticsearch data in both regions.
```

#### WebApps ![passive](https://img.shields.io/badge/passive-orange?style=for-the-badge)

WebApps are an **active-passive** component and in its current state also not highly available since we're limited by the included exporters that would cause data issues when running multiple instances at the same time.

> Per region, one WebApp instance is required for the exporter to run.

Reviewer: ```suggestion
In every region, there can be only a single WebApp instance running (i.e. in Region 0, 1 instance of Operate, 1 instance of Tasklist; same in Region 1)
```

This means that one instance will be actively serving traffic, while the other one is on standby.

##### Operate

> In the event of a total primary region loss, the following data will be lost:

Reviewer: I think it's about a chance of losing it, not guaranteed a loss:

```suggestion
In the event of a total primary region loss, the following data can be lost:
```

Author: if your primary region is lost the data is lost since it's solely in that region.
That's why Thorben is working on moving that to the engine so the data is replicated as well.

- Batch Operations <!-- Do we have a docs link for this? -->

##### Tasklist

> In the event of a total primary region loss, the following data will be lost:

Reviewer: same here:
```suggestion
In the event of a total primary region loss, the following data can be lost:
```

Author: same as above, the data will be lost.
Task assignments etc. are not replicated, otherwise you could send traffic to any Tasklist instance but it's only within that Elasticsearch of the region.

- Assignments of tasks

## Requirements

- Camunda 8
  - Minimum [helm chart version](https://github.com/camunda/camunda-platform-helm) **9.3+**
  - Minimum component images
    - Elasticsearch **8.9+**
    - Operate **8.5+**
    - Tasklist **8.5+**
    - Zeebe **8.5+**
    - Zeebe Gateway **8.5+**

> - Two connected Kubernetes clusters

Reviewer: also not sure about the term "connected" here: shall we more explicitly say what it means?

> - Two connected Kubernetes clusters

Author: suggested change:
```
- Two Kubernetes clusters
  - those need to be able to contact each other via, for example, VPC peering or similar means
  - an [example implementation](<!-- TODO: Link -->) of two VPC peered Kubernetes clusters based on AWS EKS
```

similar means could be service meshes, just not tested whether it works.

- Maximum latency of 100ms between the two Kubernetes clusters
- Open Ports between the two Kubernetes clusters
  - **9200** for Elasticsearch since Zeebe has to push data cross region
  - **26500** for the Zeebe gateway from client/worker communication
  - **26501** for the Zeebe brokers and Zeebe gateway communication
  - **26502** for the Zeebe brokers and Zeebe gateway communication

## Limitations

- This is only about a dual-region setup and does not replicate any multi-region setup with more than two Kubernetes clusters
  - a multi-region setup of 3+ clusters will behave differently due to for example the raft protocol within Zeebe
- Camunda 8 must be installed with the [Camunda Helm chart](https://github.com/camunda/camunda-platform-helm)

>   - a docker installation or plain JAR installation is not supported

Author: but fine leaving it out, just as a FYI that there are people doing that and with the mono repo probably becoming even more prominent.

- Looking at the whole Camunda Platform, it's **active-passive**, while some key components are active-active
  - meaning there's always one primary and one secondary region for serving active user traffic

>   - serving traffic to both regions will result in a detachment of the WebApps and users seeing different results in Operate and Tasklist

Author: I see what you mean but as soon as you have user traffic going to both and doing any operation it diverges. If they just look and not touch, nothing happens.

>   - serving traffic to both regions will result in a detachment of the WebApps and users seeing different results in Operate and Tasklist

Author: fine taking in the suggestion.

- The user is responsible for detecting a regional failure and executing the [operational procedure](<-- TODO: link -->)
- Identity is not supported
  - Multi-tenancy does not work
  - Role Based Access Control (RBAC) does not work
  - the reasoning behind it is that Identity brings its own problems along that require more work on our side
    - those namely being a multi-region PostgreSQL database and aligning cluster secrets of identity to ensure RBAC works properly across regions
- Optimize is not supported
  - this is due to strict coupling with Identity
- Connectors are not supported

>   - this is due to its coupling with Operate, which has a negative side effect on inbound connectors and can result in race conditions

Author: should we remove this line considering we did the same for Identity?

> Let's remove this line, we should not mention problems as the reasoning.

>   - this is due to its coupling with Operate, which has a negative side effect on inbound connectors and can result in race conditions

Author: @maxdanilov , here's still the question whether we keep it or remove it and just say not supported, no reason given.

- Only a maximum latency of 100 ms between the regions is supported

> - Only specific combinations of Zeebe broker counts and replications factors are supported <!-- TODO: Maybe Deepthi can elaborate on this and how a customer can figure out a setup -->

Author: moved it there but yeah, probably someone with more knowledge as the `todo` mentions should probably define that.

> - Only specific combinations of Zeebe broker counts and replications factors are supported <!-- TODO: Maybe Deepthi can elaborate on this and how a customer can figure out a setup -->

Author: Talked to Deepthi and will add the content later.

> - During the fallback procedure, there’s a small chance that some data will be lost for the WebApps

Author: that's actually a completely different point than the batch operations and task assignments.
Due to pointer jumps you could simply loose Zeebe data to a degree of the pointer jump. How much that is, no idea.
Maybe something for Deepthi to elaborate on.

> - During the fallback procedure, there’s a small chance that some data will be lost for the WebApps

Author: Talked to Deepthi and will add the content later.

  - Due to the difference in sequence position tracking for exporters to different Elasticsearch locations <!-- TODO: Maybe for Deepthi to elaborate furthe r -->
- Zeebe automatic cluster scaling
  - consider the cluster to be static

> - Web-Modeler is a standalone component and will not be considered further.

Author: I think the reasoning is always that you should run it outside of your automation cluster but nothing that is documented anywhere, just consulting advice.
You could have it running as multi-region app but run into the same issues as Identity due to Postgres.

  - Modeling applications can operate independently outside of the automation clusters.

> - Following topics are out of scope and have not yet been explored further and may or may not work well with the dual-region setup

Author: I'd mention service meshes since that's something you'll naturally diverge to for setting up your multi-cluster.
The DNS chaining approach doesn't scale well and is probably something you won't find a lot in production.

  - managed OpenSearch usage
  - OpenShift support
  - Service meshes

<!-- TODO: Should there be a disclaimer on we're trying to improve the support of Identity, ... with the next version? -->

## Considerations

Multi-region setups in itself bring their own complexity:

- Managing multiple Kubernetes clusters and their deployments across regions
- Monitoring and alerting
- Increased costs of multiple clusters and cross-region traffic

> - Data consistency and synchronization challenges

Author: just a side note since I see a lot of `e.g.` usage, DevEx published a [styling update](https://camunda.slack.com/archives/C026U8GBNSW/p1709922869101609) a while ago to not use latin abbreviations anymore.

#### Elasticsearch

We treat Elasticsearch as an **active-passive** component. While it may be possible to deploy it in active-active mode and stretch across regions, it's not officially documented by Elasticsearch. Such a setup generally brings its own challenges along that we can't control and may negatively impact the performance of Zeebe.

We go with the approach of having one Elasticsearch per region and configuring Zeebe to export the Elasticsearch data in both regions.

#### WebApps

WebApps are an **active-passive** component and in its current state also not highly available since we're limited by the included exporters that would cause data issues when running multiple instances at the same time.

In every region, there can be only a single WebApp instance running (i.e. in Region 0, 1 instance of Operate, 1 instance of Tasklist; same in Region 1)

This means that one instance will be actively serving traffic, while the other one is on standby.

##### Operate

In the event of a total active region loss, the following data will be lost:

> - Batch Operations <!-- Do we have a docs link for this? -->

Reviewer: should this be "uncompleted batch operations"?

##### Tasklist

In the event of a total active region loss, the following data will be lost:

> - Assignments of tasks

Reviewer: should this be "assigments of uncompleted tasks"? As afaik the completed ones are preserved in Zeebe.

## Requirements

- Camunda 8
  - Minimum [Helm chart version](https://github.com/camunda/camunda-platform-helm) **9.3+**
  - Minimum component images
    - Elasticsearch **8.9+**
    - Operate **8.5+**
    - Tasklist **8.5+**
    - Zeebe **8.5+**
    - Zeebe Gateway **8.5+**
- Two Kubernetes clusters
  - OpenShift is not supported

>   - those need to be able to contact each other via, for example, VPC peering or similar means

Reviewer: ```suggestion
  - the Kubernetes clusters need to be able to connect to each other (e.g. via VPC peering or similar means)
````

>   - an [example implementation](<!-- TODO: Link -->) of two VPC peered Kubernetes clusters based on AWS EKS

Reviewer: increasing the padding to place it as a sub-item for the Cluster list item above:
```suggestion
    - an [example implementation](<!-- TODO: Link -->) of two VPC peered Kubernetes clusters based on AWS EKS
``

- Elasticsearch usage

>   - OpenSearch (both managed and self-managed) is not supported

Reviewer: I suggest moving it up as a sub-item for `Elasticsearch 8.9+` above.

> - Maximum RTT (round trip time) of 100 ms between the two Kubernetes clusters

Reviewer: ```suggestion
- Maximum RTT (round-trip time) of 100 ms between the two Kubernetes clusters
```

Reviewer: I'd also move this as a sub-point under the `Two Kubernetes clusters` item (to have all k8s requirements in one place)

- Open Ports between the two Kubernetes clusters

>   - **9200** for Elasticsearch since Zeebe has to push data cross-region

Reviewer: ```suggestion
  - **9200** for Elasticsearch for Zeebe to push data cross-region
```

>   - **26500** for the Zeebe gateway from client/worker communication

Reviewer: ```suggestion
  - **26500** for communication to the Zeebe gateway from client/workers
```

  - **26501** for the Zeebe brokers and Zeebe gateway communication
  - **26502** for the Zeebe brokers and Zeebe gateway communication
- Only specific combinations of Zeebe broker counts and replication factors are supported

>   - `clusterSize` must be a multiple of **two** and a minimum of **four** to evenly distribute the brokers across the two regions

Reviewer: It's better to spell the numbers consistently as numbers (like we do for the port numbers):
```suggestion
  - `clusterSize` must be a multiple of **2** and a minimum of **4** to evenly distribute the brokers across the two regions
```

>   - `replicationFactor` must be **four** to ensure that the partitions are evenly distributed across the two regions

Reviewer: ```suggestion
  - `replicationFactor` must be **4** to ensure that the partitions are evenly distributed across the two regions
````

  - `partitionCount` is not restricted and depends on your workload requirements, consider having a look at [understanding sizing and scalability behavior](../../../components/best-practices/architecture/sizing-your-environment.md#understanding-sizing-and-scalability-behavior)

>   - for further information and visualization of the partition distribution consider conducting the documentation about [partitions](../../../components/zeebe/technical-concepts/partitions.md)

Reviewer: guess it makes sense to indent it under the `partitionCount`:
```suggestion
    - for further information and visualization of the partition distribution consider conducting the documentation about [partitions](../../../components/zeebe/technical-concepts/partitions.md)
```

## Limitations

> - Camunda 8 must be installed with the [Camunda Helm chart](https://github.com/camunda/camunda-platform-helm)

Reviewer: maybe an internal docs link is better: http://localhost:3000/docs/next/self-managed/platform-deployment/helm-kubernetes/overview/#use-helm-to-install-on-kubernetes

  - alternative installation methods (for example with docker-compose installation) are not supported
- Looking at the whole Camunda Platform, it's **active-passive**, while some key components are active-active
  - meaning there's always one active and one passive region for serving active user traffic
  - serving traffic to both regions will result in a detachment of the WebApps and users potentially observing different data in Operate and Tasklist

> - The customers operating their Camunda 8 setup are responsible for detecting a regional failure and executing the [operational procedure](<-- TODO: link -->)

Reviewer: I think it's rather for Requirements (and deserves an extra mention in the `Disclaimer` on the top imo)

Author: added a disclaimer and moved on top

- Identity is not supported
  - Multi-tenancy does not work
  - Role Based Access Control (RBAC) does not work
- Optimize is not supported
  - This is due to Optimize depending on Identity to work
- Connectors are not supported
  - This is due to Connectors depending on Operate to work for inbound connectors and potentially resulting in race condition
- During the fallback procedure, there’s a small chance that some data will be lost in Elasticsearch affecting Operate and Tasklist
  - This **does not** affect the processing of process instances in any way. The impact is that some information about the affected instances might not be visible in Operate and Tasklist.

>   - This will be further explained in the [operational procedure](<!-- TODO: Link -->) during the relevant step.

Reviewer: ```suggestion
  - This is be further explained in the [operational procedure](<!-- TODO: Link -->) during the relevant step.
```

- Zeebe cluster scaling must be disabled
  - Zeebe cluster size (broker count) must be static in size
- Web-Modeler is a standalone component and is not covered in this guide
  - Modeling applications can operate independently outside of the automation clusters.

> - Service meshes are presently unsupported, and we advise against their use for the setup.

Reviewer: ```suggestion
- Kubernetes service meshes are currently unsupported, and we advise against their use for the setup.
```

## Considerations

Multi-region setups in itself bring their own complexity. The following items are such complexities and are not considered in our guides.
You should familiarize yourself with those before deciding to go for a dual-region setup.

- Managing multiple Kubernetes clusters and their deployments across regions
- Monitoring and alerting
- Increased costs of multiple clusters and cross-region traffic
- Data consistency and synchronization challenges (for example brought in by the increased latency)
  - bursts of increased latency can already have an impact
- Managing DNS and incoming traffic

## Region Loss

In a dual-region setup, a loss of a region will invariably affect Camunda 8, regardless of whether it's the active or passive region.

It means that the Zeebe stretch cluster will not have a quorum when half its brokers are not reachable anymore and will stop processing any new data. This will also affect the WebApps since they can not update or push new workflows. Essentially, this means that the workflow engine to halt until the region failover procedure is completed.

> The [operational procedure](<!-- TODO: link -->) looks in detail at how to temporarily recover from a region loss and ultimately how to fully reestablish the lost region. The procedure works the same way for active or passive region loss since we don't consider traffic routing (DNS) in the scenario.

Reviewer: ```suggestion
The [operational procedure](<!-- TODO: link -->) looks in detail at how to short-term recover from a region loss and how to long-term fully reestablish the lost region. The procedure works the same way for active or passive region loss since we don't consider traffic routing (DNS) in the scenario.
```

### Active Region Loss

The loss of the active region means:

- The loss of previously mentioned data in Operate and Tasklist
- Traffic is routed to the active region, which now can't be served anymore
- The workflow engine will stop processing due to the loss of the quorum

> Looking at it from a high-level point of view, the following should be considered:

Reviewer: ```suggestion
The following high-level steps need to be taken in case of the active region loss:
```

- Follow the [operational procedure](<!-- TODO: Link -->) to temporarily recover from the region loss and unblock the workflow engine
- Reroute traffic to the passive region that will now become the new active region
- Due to the loss of data in Operate and Tasklist, you'll have to:

>   - reassign tasks in Tasklist

Reviewer: ```suggestion
  - reassign uncompleted tasks in Tasklist
```

  - recreate batch operations in Operate
- Follow the [operational procedure](<!-- TODO: Link -->) to recreate a new permanent region that will become your new passive region

### Passive Region Loss

The loss of the passive region means:

- The workflow engine will stop processing due to the loss of the quorum

> Looking at it from a high-level point of view, the following should be considered:

Reviewer: ```suggestion
The following high-level steps need to be taken in case of the passive region loss:
```

- Follow the [operational procedure](<!-- TODO: Link -->) to temporarily recover from the region loss and unblock the workflow engine
- Follow the [operational procedure](<!-- TODO: Link -->) to recreate a new permanent region that will become your new passive region

Unlike the active region loss, no data will be lost, nor will any traffic require rerouting.

## Guides

> - Get yourself familiar with our [AWS setup guide](<!-- TODO: link -->) that showcases an example setup in AWS by utilizing the managed Elastic Kubernetes Service (EKS) and VPC peering for a dual-region setup with Terraform.

Reviewer: I'd also add a subitem:

```
  - The concepts in the guide are mainly cloud-agnostic and the guide can be adopted to other cloud providers.
```

:::

## Architecture

<DualRegion />

The illustrated architecture consists of two regions. Each region houses a Kubernetes cluster in which Camunda 8 is deployed. Those two Kubernetes clusters are capable of communicating with each other.

One of the regions will be considered **active** and the other one **passive**. User traffic must only reach the **active** region. We consider **Region 0**, the one underlined in green, to be the active region and **Region 1** to be the passive region. In our case, user traffic would only go to **Region 0**. **Region 1** would be considered passive and used in case of the loss of the active region. Due to Zeebe's data replication, you can recover from an active region loss by utilizing the passive region without much downtime.

Zeebe stretches across the regions due to its use of the [Raft protocol](<https://en.wikipedia.org/wiki/Raft_(algorithm)>) allowing it to communicate and replicate data between all brokers across. Zeebe is exporting data to two Elasticsearch instances, one in each region. Operate and Tasklist will import the previously exported data and run per region.

The currently supported Camunda 8 components are the following:

> - Zeebe (Workflow Engine)

Reviewer 2: Should likely mention the others as well, such as Connectors, Optimize, Identity, etc.
